### PR TITLE
fix: merge control markdown with json (#1528)

### PR DESCRIPTION
**Pull request by d10n**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -259,7 +259,9 @@ def tmp_empty_cwd(tmp_path: pathlib.Path) -> Iterator[pathlib.Path]:
 @pytest.fixture(scope='function')
 def testdata_dir() -> pathlib.Path:
     """Return absolute path to test data directory."""
-    test_data_source = pathlib.Path('tests/data')
+    test_dir = pathlib.Path(__file__).parent.resolve()
```
Comment (degenaro) [inline tests/conftest.py:262]: Why do we switch to using __file__ ?

File: tests/trestle/core/commands/validate_test.py [hunk 2]
```diff
@@ -58,8 +56,11 @@
         ('my_test_model', '-t', False), ('my_test_model', '-a', False), ('my_test_model', '-x', False)
     ]
 )
-def test_validation_happy(name, mode, parent, tmp_trestle_dir: pathlib.Path, monkeypatch: MonkeyPatch) -> None:
+def test_validation_happy(
+    name, mode, parent, tmp_trestle_dir: pathlib.Path, testdata_dir: pathlib.Path, monkeypatch: MonkeyPatch
+) -> None:
     """Test successful validation runs."""
+    test_data_dir = testdata_dir
```
Comment (degenaro) [inline tests/trestle/core/commands/validate_test.py:63]: Why not use test_data_dir as the parameter name and remove this statement?

**Review by degenaro — CHANGES_REQUESTED** [review]
Is there a test case that demonstrates the failure before this PR that now passes?

File: tests/conftest.py (hunk 1 above)
Comment (d10n) [reply tests/conftest.py:262]: Expanding on the commit message for the line: to enable running / debugging unit tests in PyCharm; PyCharm sets working directories for the test runner incorrectly, even when manually specified. This is a PyCharm problem, but enabling tools is good, so why not? 😃 It also enables running unit tests from any directory in the shell.

File: tests/trestle/core/commands/validate_test.py [hunk 3]
```diff
@@ -101,9 +102,10 @@ def test_validation_happy(name, mode, parent, tmp_trestle_dir: pathlib.Path, mon
     ]
 )
 def test_validation_unhappy(
-    name, mode, parent, status, tmp_trestle_dir: pathlib.Path, monkeypatch: MonkeyPatch
+    name, mode, parent, status, tmp_trestle_dir: pathlib.Path, testdata_dir: pathlib.Path, monkeypatch: MonkeyPatch
 ) -> None:
     """Test failure modes of validation."""
+    test_data_dir = testdata_dir
```
[the remark this answers is not in the document]
Comment (d10n) [reply tests/trestle/core/commands/validate_test.py:108]: I agree this is messy. I did this to keep the changed line count small, and I was planning to do more test clean-up in a dedicated PR, but I can fix this now.

**Comment by degenaro** [comment]
`In Python, __file__ is a special built-in variable that represents the path to the current script file.` I still struggle with why we want to change?

**Comment by d10n** [comment]
We discussed the unit test updates in Slack, and I am making a separate PR with them.

**Review by degenaro — APPROVED** [review]
LGTM. 

Are PR & issue linked?

**Review by vikas-agarwal76 — APPROVED** [review]
LGTM

**Pull request merged**